Assets/Plugins/iOS/BidMachineUnityBridge.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  BidMachineiOSBridge.swift
+//  UnityFramework
+//
+//  Created by Dzmitry on 27/09/2024.
+//
+
+import Foundation
+import BidMachine
+
+let iOSUnityBridge = BidMachineUnityBridge(instance: .shared)
+
+final class BidMachineUnityBridge {
+    let rewardedBridge: RewardedAdBridge
+
+    var isInitialized: Bool {
+        instance.isInitialized
+    }
+    
+    private let instance: BidMachineSdk
+    
+    init(instance: BidMachineSdk) {
+        self.instance = instance
+        self.rewardedBridge = RewardedAdBridge(instance: instance)
+    }
+    
+    func initialize(with sourceId: String) {
+        instance.initializeSdk(sourceId)
+    }
+
+    func setEndpoint(url: String) {
+        instance.installFeature(Constant.endpointKey, url)
+    }
+    
+    func setLoggingEnabled(_ enabled: Bool) {
+        instance.populate {
+            $0.withLoggingMode(enabled)
+        }
+    }
+    
+    func setTestEnabled(_ enabled: Bool) {
+        instance.populate {
+            $0.withTestMode(enabled)
+        }
+    }
+    
+    func setTargeting(parameters: TargetingParameters) {
+        instance.targetingInfo.populate { info in
```

Comment:
As minor way to improve you can make extensions for builders:
```swift
extension BidMachineTargetingInfoBuilderProtocol {
    @discardableResult
    func withTargetingParameters(_ parameters: TargetingParameters) -> Self {
        // All logic here
        return self
    }
}
```
in `TargetingParameters` file.

And use this extensions:
```swift 
instance.targetingInfo.populate { info in
    info.withTargetingParameters(parameters)
}
```

Reply:
✅ implemented proposed solution